server/src/com/thoughtworks/go/server/service/GoConfigService.java

```diff
@@ -147,6 +147,15 @@ private boolean canEditPipeline(String pipelineName, Username username, Localize
         return canEditPipeline(pipelineName, username, result, findGroupNameByPipeline(new CaseInsensitiveString(pipelineName)));
     }
 
+    public boolean canEditPipeline(String pipelineName, Username username) {
+        if (getCurrentConfig().hasPipelineNamed(new CaseInsensitiveString(pipelineName)) &&
```

Comment:
`getCurrentConfig().hasPipelineNamed` internally uses `getPipelineConfigByName` which goes over the list of pipeline-groups and hence could become too slow. Could you change the implementation of `hasPipelineNamed` to use `pipelineConfigByName`. `pipelineConfigByName` uses a cached map to get this information.